sklearn/decomposition/pca.py

```diff
@@ -177,35 +212,60 @@ class PCA(BaseEstimator, TransformerMixin):
     via the score and score_samples methods.
     See http://www.miketipping.com/papers/met-mppca.pdf
 
-    Due to implementation subtleties of the Singular Value Decomposition (SVD),
-    which is used in this implementation, running fit twice on the same matrix
-    can lead to principal components with signs flipped (change in direction).
-    For this reason, it is important to always use the same estimator object to
-    transform data in a consistent fashion.
+    For svd_solver == 'arpack', refer to `scipy.sparse.linalg.svds`.
+
+    For svd_solver == 'randomized', see:
+    `Finding structure with randomness: Stochastic algorithms
+    for constructing approximate matrix decompositions Halko, et al., 2009
+    (arXiv:909)`
+    `A randomized algorithm for the decomposition of matrices
+    Per-Gunnar Martinsson, Vladimir Rokhlin and Mark Tygert`
+
 
     Examples
     --------
-
     >>> import numpy as np
     >>> from sklearn.decomposition import PCA
     >>> X = np.array([[-1, -1], [-2, -1], [-3, -2], [1, 1], [2, 1], [3, 2]])
     >>> pca = PCA(n_components=2)
     >>> pca.fit(X)
-    PCA(copy=True, n_components=2, whiten=False)
+    PCA(copy=True, iterated_power=3, n_components=2, random_state=None,
+      svd_solver='auto', tol=0.0, whiten=False)
     >>> print(pca.explained_variance_ratio_) # doctest: +ELLIPSIS
     [ 0.99244...  0.00755...]
 
+    >>> pca = PCA(n_components=2, svd_solver='full')
+    >>> pca.fit(X)                 # doctest: +ELLIPSIS +NORMALIZE_WHITESPACE
+    PCA(copy=True, iterated_power=3, n_components=2, random_state=None,
+      svd_solver='full', tol=0.0, whiten=False)
+    >>> print(pca.explained_variance_ratio_) # doctest: +ELLIPSIS
+    [ 0.99244...  0.00755...]
+
+    >>> pca = PCA(n_components=1, svd_solver='arpack')
+    >>> pca.fit(X)
+    PCA(copy=True, iterated_power=3, n_components=1, random_state=None,
+      svd_solver='arpack', tol=0.0, whiten=False)
+    >>> print(pca.explained_variance_ratio_) # doctest: +ELLIPSIS
+    [ 0.99244...]
+
     See also
     --------
-    RandomizedPCA
     KernelPCA
     SparsePCA
     TruncatedSVD
+    IncrementalPCA
     """
-    def __init__(self, n_components=None, copy=True, whiten=False):
+
+    def __init__(self, n_components=None, copy=True, whiten=False,
+                 svd_solver='auto', tol=0.0, iterated_power=3,
```

Comment:
we use 2 power iterations as the default for randomized_svd and 3 for PCA. I wonder if we should not use 2 for PCA as well or change the default of randomized_svd to 3.